File: android/src/main/java/com/getcapacitor/community/NearbyConnectionsConfig.java
package com.getcapacitor.community;

import static com.getcapacitor.community.NearbyConnectionsHelper.toConnectionType;
import static com.getcapacitor.community.NearbyConnectionsHelper.toStrategy;

import android.util.Base64;
import androidx.annotation.Nullable;
import com.google.android.gms.nearby.connection.Payload;
import com.google.android.gms.nearby.connection.Strategy;

public class NearbyConnectionsConfig {

    @Nullable
    private String endpointName;

    @Nullable
    private String serviceID;

    @Nullable
    private Strategy strategy;

    @Nullable
    private Boolean lowPower;

    @Nullable
    private Integer connectionType;

    @Nullable
    private Boolean autoConnect;

    @Nullable
    private Payload payload;

    public void setEndpointName(@Nullable String endpointName) {
        this.endpointName = endpointName;
    }

    public void setServiceID(@Nullable String serviceID) {
        this.serviceID = serviceID;
    }

    public void setStrategy(@Nullable String strategy) {
        this.strategy = toStrategy(strategy);
    }

    public void setLowPower(@Nullable Boolean lowPower) {
        this.lowPower = lowPower;
    }

    public void setConnectionType(@Nullable Integer connectionType) {
        this.connectionType = connectionType;
    }

    public void setConnectionType(@Nullable String connectionType) {
        this.connectionType = toConnectionType(connectionType);
    }

    public void setAutoConnect(@Nullable Boolean autoConnect) {
        this.autoConnect = autoConnect;
    }

    public void setPayload(@Nullable String payload) {
        this.payload = (payload == null) ? null : Payload.fromBytes(Base64.decode(payload, Base64.NO_WRAP));
    }

    @Nullable
    public String getEndpointName() {
        return endpointName;
    }

    @Nullable
    public String getServiceID() {
        return serviceID;
    }

    @Nullable
    public Strategy getStrategy() {
        return strategy;
    }

    @Nullable
    public Boolean getLowPower() {
        return lowPower;
    }

    @Nullable
    public Integer getConnectionType() {
        return connectionType;
    }

    @Nullable
    public Boolean getAutoConnect() {
        return autoConnect;
    }

    @Nullable
    public Payload getPayload() {
        return payload;
    }
}

File: ios/Sources/NearbyConnectionsPlugin/NearbyConnectionsConfig.swift
import Capacitor

public class NearbyConnectionsConfig {
    private var endpointName: String?

    private var serviceID: String?
    private var strategy: String?

    private var autoConnect: Bool?
    private var payload: String?

    init(config: PluginConfig) {
        self.endpointName = config.getString("endpointName")

        self.serviceID = config.getString("serviceID")
        self.strategy = config.getString("strategy")

        if config.getString("autoConnect") != nil {
            self.autoConnect = config.getBoolean("autoConnect", false)
        }
        self.payload = config.getString("payload")
    }

    func setEndpointName(_ endpointName: String?) {
        self.endpointName = endpointName
    }

    func setServiceID(_ serviceID: String?) {
        self.serviceID = serviceID
    }
    func setStrategy(_ strategy: String?) {
        self.strategy = strategy
    }

    func setAutoConnect(_ autoConnect: Bool?) {
        self.autoConnect = autoConnect
    }
    func setPayload(_ payload: String?) {
        self.payload = payload
    }

    func getEndpointName() -> String? {
        return self.endpointName
    }

    func getServiceID() -> String? {
        return self.serviceID
    }
    func getStrategy() -> String? {
        return self.strategy
    }

    func getAutoConnect() -> Bool? {
        return self.autoConnect
    }
    func getPayload() -> String? {
        return self.payload
    }
}
